tests/Resource/NotificationPreferencesTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Moip\Tests\Resource;
+
+use Moip\Tests\TestCase;
+use Moip\Resource\MoipResource;
+use Moip\Resource\NotificationPreferences;
+
+/**
+ * Description of NotificationPreferencesTest.
+ */
+class NotificationPreferencesTest extends TestCase
+{
+    private function createNotification()
+    {
+        $this->mockHttpSession($this->body_notification_preference);
+        $notification = $this->moip->notifications()->addEvent('ORDER.*')
+            ->addEvent('PAYMENT.AUTHORIZED')
+            ->setTarget('http://requestb.in/1dhjesw1')
+            ->create();
+
+        return $notification;
+    }
+
+    public function testShouldCreateNotificationPreference()
+    {
+        $notification = $this->createNotification();
+        $this->assertNotEmpty($notification->getId());
+    }
+
+    public function testEndpointDeleteNotificationPreference()
+    {
```

Comment:
Hmm...This test doesn't look great to me.
I feel like `generatePath` should be a method of `MoipResource` and you would test it there. 
Testing on the parent class should be enough :)